tests/unit/planning/test_trajectory_validation.py

```
@@ -0,0 +1,255 @@
+    assert leg.repair is LegRepair.CLEAN
+    assert leg.ok is True
+    assert leg.repaired is False
+
+
+def test_plan_clear_leg_resmoothed_subdivides_and_clears() -> None:
+    a = Pose(0.0, 0.0, 0.0)
+    b = Pose(10.0, 0.0, 0.0)
+    maneuver = ManeuverType.COVERAGE
+    midpoint = _chord_midpoint(a, b)
+    motion = _scaled_plan_motion(scale=10.0)
+    env = FakeEnv(lambda path, step_m, call_no: len(path.segments) >= 2 or path.total_length_m <= 60.0)
```

Ruff (0.16.3)

[warning] Unused lambda argument: `step_m`

(ARG005)